Source/Csla.Channels.RabbitMq/RabbitMqPortal.cs

```diff
@@ -21,54 +21,25 @@ namespace Csla.Channels.RabbitMq
   /// <summary>
   /// Exposes server-side DataPortal functionality through RabbitMQ
   /// </summary>
+  /// <remarks>
+  /// Creates an instance of the type
+  /// </remarks>
   public class RabbitMqPortal : IDisposable
   {
-    private IDataPortalServer dataPortalServer;
-    private ApplicationContext _applicationContext;
-
-    /// <summary>
-    /// Creates an instance of the type
-    /// </summary>
-    /// <param name="applicationContext"></param>
-    /// <param name="dataPortal">Data portal server service</param>
-    /// <exception cref="ArgumentNullException"><paramref name="applicationContext"/> or <paramref name="dataPortal"/> is <see langword="null"/>.</exception>
-    public RabbitMqPortal(ApplicationContext applicationContext, IDataPortalServer dataPortal)
+    internal RabbitMqPortal(ApplicationContext applicationContext, IDataPortalServer dataPortal, RabbitMqPortalOptions rabbitMqPortalOptions)
     {
-      dataPortalServer = dataPortal ?? throw new ArgumentNullException(nameof(dataPortal));
+      _dataPortalServer = dataPortal ?? throw new ArgumentNullException(nameof(dataPortal));
       _applicationContext = applicationContext ?? throw new ArgumentNullException(nameof(applicationContext));
+      DataPortalUri = rabbitMqPortalOptions.DataPortalUri;
     }
 
-    /// <summary>
-    /// Creates an instance of the object.
-    /// </summary>
-    /// <param name="applicationContext"></param>
-    /// <param name="dataPortal">Data portal server service</param>
-    /// <param name="dataPortalUrl">URI for the data portal</param>
-    /// <exception cref="ArgumentException"><paramref name="dataPortalUrl"/> is null, empty or only consists of white spaces.</exception>
-    public RabbitMqPortal(ApplicationContext applicationContext, IDataPortalServer dataPortal, string dataPortalUrl)
-      : this(applicationContext, dataPortal)
-    {
-      if (string.IsNullOrWhiteSpace(dataPortalUrl))
-        throw new ArgumentException(string.Format(Properties.Resources.StringNotNullOrWhiteSpaceException, nameof(dataPortalUrl)), nameof(dataPortalUrl));
-
-      DataPortalUrl = dataPortalUrl;
-    }
-
-    /// <summary>
-    /// Gets the URI for the data portal service.
-    /// </summary>
-    public string DataPortalUrl { get; } = string.Empty;
+    private readonly IDataPortalServer _dataPortalServer;
+    private readonly ApplicationContext _applicationContext;
 
     private IConnection? Connection;
     private IModel? Channel;
     private string? DataPortalQueueName;
 
-    /// <summary>
-    /// Gets or sets the timeout for network
-    /// operations in seconds (default is 30 seconds).
-    /// </summary>
-    public int Timeout { get; set; } = 30;
-
     private Uri? DataPortalUri { get; set; }
```

Comment:
Should we still keep this nullable and not catch that during configuration/instantiation?
We should change from runtime error to app startup error to catch that earlier.